bin/picom-trans

```diff
@@ -46,194 +49,278 @@ print_usage() {
-  *compton-trans*)  echo "Warning: compton has been renamed, please use picom-trans instead" >& 2;;
-esac
 
-# "command" is a shell built-in, faster than "which"
-if test -z "$(command -v xprop)" -o -z "$(command -v xwininfo)"; then
-  echo 'The command xwininfo or xprop is not available. They might reside in a package named xwininfo, xprop, x11-utils, xorg-xprop, or xorg-xwininfo.' >& 2
-  exit 1
-fi
+function parse_args
+{ #{
 
-if test $# -eq 0; then
-  print_usage >& 2
-  exit 1
-fi
+  # Iterate over every option and convert
+  # longopts (GNU) into POSIX-compatible opts
+  i=1       # because we start from "$1", not from "$0"
+  while [ $i -le $# ]
+  do
+    # Catch invalid options
+    case "$1" in
+      (--opacity=|--name=|--window=)
+        echo "Option $1 needs a value." >&2
+        exit 1
+    esac
 
-# Variables
-active=
-wprefix=
-window=
-opacity=
-cur=
-action=
-treeout=
-wid=
-topmost=
-lineno=
-option=
-v=
-
-# We make getopts stop on any argument it doesn't recognize
-# or errors on. This allows for things like `picom-trans -5`
-# as well as `picom-trans -c +5 -s` (contrived example).
-while test $# -gt 0; do
-  # Reset option index
-  OPTIND=1
-
-  # Read options
-  while getopts ':hscrtdgn:w:o:-:' option "$@"; do
-    if test "$option" = '-'; then
-      case "$OPTARG" in
-        help | select | current | reset | toggle | delete | get)
-          v=''
-        ;;
-        name | window | opacity)
-          eval v=\$$OPTIND
-          OPTIND=$((OPTIND + 1))
-        ;;
-        name=* | window=* | opacity=*)
-          v=$(echo "$OPTARG" | sed -E 's/^[^=]+=//')
-        ;;
-        *)
-          echo "$0: illegal option $OPTARG" >& 2
-          exit 1
-        ;;
-      esac
-      option=$(echo "$OPTARG" | cut -c 1)
-      OPTARG=$v
-    fi
-    case "$option" in
-      h) print_usage; exit 0 ;;
-      s) wprefix=''; window='' ;;
-      c)
-        active=$(xprop -root -notype _NET_ACTIVE_WINDOW \
-          | grep -Eo '0x[[:xdigit:]]+' | head -n 1)
-        wprefix='-id'; window=$active
-      ;;
-      r) action='reset' ;;
-      t) action='toggle' ;;
-      d) action='delete' ;;
-      g) action='get' ;;
-      n) wprefix='-name'; window=$OPTARG ;;
-      w) wprefix='-id'; window=$OPTARG ;;
-      o) opacity=$OPTARG ;;
-      \?) break ;;
+    # Separate  "--ARG=VAL"  into  "--ARG" "VAL"
+    case "$1" in
+      (--opacity=*|--name=*|--window=*)
+        ARG="$(echo "$1" | sed -E 's/(--[^=]+)=.*$/\1/')"
+        VAL="${1##${ARG}=}"
+        shift && set -- "$ARG" "$VAL" "$@"
     esac
-  done
 
-  # Read positional arguments
-  shift $((OPTIND - 1))
-  test -n "$1" && opacity=$1 && shift
-done
+    # Turn into short form
+    case "$1" in
+      (--help|--opacity|--get|--delete|--toggle|--reset|--select|--current|--name|--window)
+        ARG=${1#-}                      # remove one '-' from prefix
+        ARG="$(echo $ARG | cut -c -2)"  # get first two characters
+        shift && set -- "$ARG" "$@"
+    esac
```

Comment:
This might need an additional step as it currently fails with the non-descriptive `getopts` error `illegal option -- -` for invalid long-options.

Reply:
Great catch!  
Fixed.